src/data/sigma/basic.lean

```diff
@@ -80,6 +80,14 @@ begin
   exact ⟨⟨i, x⟩, rfl⟩
 end
 
+/-- like `function.curry` with the difference that `curry` produces a function with a dependent type -/
```

Comment:
Please describe the new function instead of comparing it to an existing one. I'm not sure which one `curry` refers to here. Same for `uncurry` below.